Helper/Data.php

```diff
@@ -535,26 +533,42 @@ public function rebuildStoreProductIndexPage($storeId, $collectionDefault, $page
         $this->logger->log('Loaded ' . count($collection) . ' products');
         $this->logger->stop('LOADING ' . $this->logger->getStoreName($storeId) . ' collection page ' . $page . ', pageSize ' . $pageSize);
 
-        $index_name = $this->productHelper->getIndexName($storeId, $useTmpIndex);
+        $indexName = $this->productHelper->getIndexName($storeId, $useTmpIndex);
 
         $indexData = $this->getProductsRecords($storeId, $collection, $productIds);
 
         if (!empty($indexData['toIndex'])) {
             $this->logger->start('ADD/UPDATE TO ALGOLIA');
 
-            $this->algoliaHelper->addObjects($indexData['toIndex'], $index_name);
+            $this->algoliaHelper->addObjects($indexData['toIndex'], $indexName);
 
             $this->logger->log('Product IDs: ' . implode(', ', array_keys($indexData['toIndex'])));
             $this->logger->stop('ADD/UPDATE TO ALGOLIA');
         }
 
         if (!empty($indexData['toRemove'])) {
-            $this->logger->start('REMOVE FROM ALGOLIA');
+            $toRealRemove = [];
+
+            if (count($indexData['toRemove']) === 1) {
+                $toRealRemove = $indexData['toRemove'];
+            } else {
+                $indexData['toRemove'] = array_map('strval', $indexData['toRemove']);
+                $objects = $this->algoliaHelper->getObjects($indexName, $indexData['toRemove']);
```

Comment:
No, it's in algoliaHelper - https://github.com/algolia/algoliasearch-magento-2/blob/master/Helper/AlgoliaHelper.php#L69